homeassistant/util/unit_system.py

```diff
@@ -100,12 +100,35 @@ def temperature(self: object, temperature: float, from_unit: str) -> float:
                                         from_unit, self.temperature_unit)
 
     def length(self: object, length: float, from_unit: str) -> float:
-        """Convert the given length to this unit system."""
+        """Convert the given length to this unit system return a float."""
+        converted = self.length_with_display_obj(length, from_unit)
```

Comment:
This is a bug as it can now return meters instead of kilometers. You need to keep this the old implementation.